pkg/build/strategies/sti/sti.go

```diff
@@ -253,7 +260,7 @@ func (b *STI) Determine(request *api.Request) (err error) {
 	// we're assuming save-artifacts to exists for embedded scripts (if not we'll
 	// warn a user upon container failure and proceed with clean build)
 	// for external save-artifacts - check its existence
-	saveArtifactsExists := request.ExternalOptionalScripts ||
+	saveArtifactsExists := !b.externalScripts[api.SaveArtifacts] ||
```

Comment:
I still don't understand why the first half of this check is needed?  isn't it always possible save-artifacts exists inside the image, regardless of what was downloaded?